quickwit/quickwit-doc-mapper/src/default_doc_mapper/default_mapper.rs

```diff
@@ -320,6 +312,52 @@ impl TryFrom<DefaultDocMapperBuilder> for DefaultDocMapper {
     }
 }
 
+fn resolve_default_search_fields(
+    default_search_fields: &[String],
+    schema: &Schema,
+) -> Result<Vec<String>, anyhow::Error> {
+    let mut default_search_field_names = Vec::new();
+    for field_name in default_search_fields {
+        if default_search_field_names.contains(field_name) {
+            bail!("Duplicated default search field: `{}`", field_name)
+        }
+        if field_name.contains('.') {
+            let potential_json_field_name = field_name
```

Comment:
Quickwit has a weird feature where you can start schemaless and add some schema specification for some stuff later.

This works by having a JSON field called `product_details` for instance, and a field called `product_details.price` (For instance to specify it as a fast field).

On the search side, if you want to identify the field that is targetted, you need to pick the longest field name.
so product_details.price targets the u64 field in my example.

Your implementation should replicate this logic.

Reply:
No you do no need to espace the . here.
I am talking of the object type
https://quickwit.io/docs/configuration/index-config#object

Let me give a concrete example:
I am using dynamic mode. I'd like to index everything I am being given. 
All of the text should be indexed with the raw tokenizer, except the  `product.description` part that should be tokenized.

I can do that today with the following doc mapper

```yaml
mode: dynamic
field_mappings:
    - name: product
      type: object
      field_mappings:
          name: description
          type: text
          tokenizer: default
```

Given a product 
```json
{"author": "user2312", "product": {"description": "cool umbrella", "id": "2222131"}}
```

In that case
```
author: is mapped in the dynamic field. the raw tokenizer is used by default
product.id is mapping in the dynamic field. the raw tokenizer is used by default
product.descirpiton is mapped in the static adhoc field and tokenizer
```

In the query parser
`product.description:red` works without escaping the `.`